web/src/settings_ui.ts

```diff
@@ -6,46 +6,63 @@ import {$t, $t_html} from "./i18n";
 import * as loading from "./loading";
 import * as ui_report from "./ui_report";
 
-export function display_checkmark($elem) {
+export function display_checkmark($elem: JQuery<HTMLElement>): void {
     const check_mark = document.createElement("img");
     check_mark.src = checkbox_image;
     $elem.prepend(check_mark);
     $(check_mark).css("width", "13px");
 }
 
-export const strings = {
+type Strings = {
+    failure_html: string;
+    saving: string;
+    success_html: string;
+};
+
+export const strings: Strings = {
     success_html: $t_html({defaultMessage: "Saved"}),
     failure_html: $t_html({defaultMessage: "Save failed"}),
     saving: $t({defaultMessage: "Saving"}),
 };
 
+type DoSettingsChangeOptions = {
+    failure_msg_html?: string;
+    sticky?: boolean;
+    success_msg_html?: string;
+    $error_msg_element?: JQuery<HTMLElement>;
+    error_continuation?(xhr?: JQuery.jqXHR): void;
+    success_continuation?(response_data?: unknown): void;
+};
```

Comment:
This type will only ever be useful once, in the annotation of `do_settings_change`. I’d prefer to inline it there, and put the fields in the same order as they’re written there.

Reply:
Okay

Reply:
@andersk I made them inline, but the order cannot be the same because typescript-eslint wants scalar fields to be defined before the functions. So, I changed the order in both the actual object and the type. I've made a new commit for it.